src/GitHubRepoManager/Functions/Public/Set-GitHubRepoVulnerabilitySetting.ps1

```diff
@@ -0,0 +1,46 @@
+<#
+.SYNOPSIS
+    A PowerShell function for enabling vulnerability alerts and automated security fixes.
+.DESCRIPTION
+    A PowerShell function for enabling vulnerability alerts and automated security fixes.
+.PARAMETER GitHubOrg
+    The name of the GitHub organisation
+.PARAMETER RepoSearchString
+    A search pattern for the repos to query.  The like comparison operator is used to match repos names so suffix any partial patterns with *
+.EXAMPLE
+    Set-GitHubSessionInformation -PatToken <not-a-real-pat-token>
+    Set-GitHubAutomatedSecurityFixes -GitHubOrg MyGitHubOrg -RepoSearchString MyRepo
+.NOTES
+    Uses:
+    https://docs.github.com/en/rest/reference/repos#enable-vulnerability-alerts
+    https://docs.github.com/en/rest/reference/repos#enable-automated-security-fixes
+    Vulnerbility alerts must be enabled before enabling automated security fixes
+#>
+function Set-GitHubRepoVulnerabilitySetting {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "RepoSearchString", Justification = "False positive as rule does not know that Where-Object operates within the same scope")]
+    [CmdletBinding(SupportsShouldProcess)]
+    Param (
+        [String]$GitHubOrg,
+        [String]$RepoSearchString
+    )
+
+    $Uri = "/orgs/$GitHubOrg/repos?type=all"
+    Write-Verbose "Searching org $GitHubOrg"
+    $Repos = Invoke-GitHubRestMethod -Method GET -Uri $uri
+    Write-Verbose -Message "Got $($Repos.Length) repositories"
+
+    $MatchedRepos = $Repos | Where-Object {$_.name -like $RepoSearchString -and $_.archived -eq $false}
```

Comment:
Using the -`match` comparison operator and a regular expression to filter by a repository prefix rather than a repo search string as implemented [here](https://github.com/SkillsFundingAgency/das-github-toolkit/blob/8ecbce3dd3152a49c51ce3d4a0e487905c16727f/src/GitHubTools/Functions/Public/Search-DotNetFramework.ps1#L36) would produce more predictable results.  eg passing in a RepoSearchString of "das" would pick up repos that started with the word "Dashboard" which would be undesirable

Reply:
Changed to `-match` and using a `RepoPrefix` parameter